mypy/checkmember.py

```diff
@@ -40,8 +40,10 @@ def analyze_member_access(name: str,
       1. lvalue or non-lvalue access (i.e. setter or getter access)
       2. supertype access (when using super(); is_super == True and
          override_info should refer to the supertype)
+
+    actual_self is the type of E in the expression E.foo
```

Comment:
Yeah, they aren't the same -- though it can appear so on a quick reading, which makes it important to document the distinction. Previously it was more obvious -- `report_type` is the type that will be reported in an error message. When initially called, `typ` is the the type of the base object (or the type of the E). In recursive calls it can be a supertype of the original value of `typ` (not necessarily a superclass). Could you add something along those lines to the docstring, even though it's not directly related to this PR? 

Finally, the most interesting question is when is `actual_self` different from `typ` on the initial (non-recursive) call and why. Maybe this only gets set in `analyze_super`? (If this isn't relevant to this PR, then this can be left out from the docstring.)